akka-actor/src/main/scala/akka/io/Inet.scala

```diff
@@ -20,23 +20,24 @@ object Inet {
     /**
      * Action to be taken for this option before bind() is called
      */
-    def beforeDatagramBind(ds: DatagramSocket): Unit = ()
+    def beforeDatagramBind(ds: DatagramSocket): Unit = ds match { case _ ⇒ () } // avoid compiler warning
```

Comment:
If we don't turn off the warning let's create a Blockhole utility to send these unwanted params to

Reply:
https://github.com/scala/bug/issues/10347

Reply:
I think those empty matches are too verbose, confusing and will only be a constant nagging on contributors (and ourselves) that forget to add them. However great it is when the compiler will find a mistake I don't think it's worth introducing so much boilerplate.

I think that silencer `@unused` annotation is needed before this becomes useful.

If we can define a blackhole syntax that is compact and zero runtime overhead it might be ok.

Reply:
```suggestion
    def beforeDatagramBind(@deprecated("unused", "") ds: DatagramSocket): Unit = ()
```